packages/cron/src/jobs/dagcargo.js

```diff
@@ -47,7 +49,12 @@ export async function updateDagSizes({ pg, after }) {
   const countRes = await pg.query(COUNT_CONTENT_WITHOUT_SIZE, [
     after.toISOString(),
   ])
-  const total = countRes.rows[0].count
+  assert.ok(countRes.rows[0])
```

Comment:
now that node.js assert.ok is typed as a tsc type guard, this is how i chose to get type safety. 

It's at risk of throwing more errors.
but less at risk of more mysterious runtime errors.

@alanshaw lmk if you'd rather I just use a type assertion (which may be incorrect at runtime) rather than the asserts.

Reply:
We have not typically used `assert` outside of tests. It seems to add a lot of noise to the code base and (at least for me) is making it quite difficult to read the actual code. It looks like you might end up asserting on types multiple times in multiple places and I think it would be more DRY to just use a `@typedef` or `@type` once. There's also probably some performance overhead we're suffering by continually type checking everything.

It definitely makes sense to use `assert` to validate types for inputs (such as in a REST endpoint, or user input from a CLI) at _runtime_ (and likely in some other situations) but I wouldn't use it like to obtain type saftey like this in your editor or for the typescript compiler.

Reply:
I will update to use less runtime assert and instead use a [type assertion](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#type-assertions) via js comment.